pkg/agents/team4/trainingAgent/generateNewBestAgents.py

```diff
@@ -2,220 +2,188 @@
 import sys
 import random
 import json
+import math
 import numpy as np
 
+# input files
 best_agents_file_name = sys.argv[1]
 agent_life_expectancies_file_name = sys.argv[2]
-number_of_health_levels = int(sys.argv[3])
-agent_death_rate_file_name = sys.argv[4]
-agent_our_life_expectancies_file_name = sys.argv[5]
-selfish_flag = sys.argv[6]
+agent_death_rate_file_name = sys.argv[3]
+agent_our_life_expectancies_file_name = sys.argv[4]
+agent_other_life_expectancies_file_name = sys.argv[5]
+
+# input flags and numerical parameters 
+number_of_health_levels = int(sys.argv[6])
+selfish_flag = sys.argv[7]
+selfless_flag = sys.argv[8]
+current_iteration = int(sys.argv[9])
+
+def generate_mutated_agent(agent, attribute, population_size): # We define a mutated angent from a select base agent in the previous iteration.
+    
+    noise_multiplier_for_attribute = {
+        "FoodToEat": math.sqrt(population_size)*5/math.sqrt(current_iteration), # Value will tend to 10
+        "WaitProbability": math.sqrt(population_size)*10/math.sqrt(current_iteration)
+    }
+
+    mu, sigma = 0, 1  # mean and standard deviation
+
+    for i in attribute:
+        for j in range(number_of_health_levels):
+            rand = int(np.random.normal(mu, noise_multiplier_for_attribute[i]*sigma))
+            agent[i][j] += rand
+            
+            if ( agent[i][j] > 60):
+                agent[i][j] = 60
+            if (agent[i][j] < 0):
+                agent[i][j] = 0        
+    return agent
+
+def generate_mixed_agent(agent_list, number_of_agents_to_use, attribute): # We define a mixed agent from parameters of the previous best agents
+
+    picked_best_agents = agent_list[:number_of_agents_to_use]
+
+    bands = []
+    for i in range(number_of_agents_to_use): # determines probability of each agent within the number of agents to use
+        bands.append(((1/number_of_agents_to_use) * (1/(i+1))))
+        
```

Comment:
MINOR: would be more readable in list comprehension `bands = [ (1/number_of_agents_to_use) * (1/(i+1))) for i in range(number_of_agents_to_use) ]`

Reply:
IMO a lot of the for-loops in this file could be condensed for readability but that's also minor